opensearch/src/error.rs

```diff
@@ -80,11 +83,32 @@ enum Kind {
     #[cfg(feature = "aws-auth")]
     #[error("AwsSigV4 error: {0}")]
     AwsSigV4(#[from] crate::http::aws_auth::AwsSigV4Error),
+
+    #[error("request initializer error: {0}")]
+    RequestInitializer(#[source] BoxError<'static>),
```

Comment:
I think we could get right of boxing (and this `BoxError`) type altogether and just use the reference `&'static (dyn std::error::Error + Sync + Send)`:

```
    #[error("request initializer error: {0}")]
    RequestInitializer(#[source] &'static (dyn std::error::Error + Sync + Send)),

    #[error("request pipeline error: {0}")]
    RequestPipeline(#[source] &'static (dyn std::error::Error + Sync + Send)),
```

It is not necessarily a better option (in my opinion) but it is a bit simpler from "less abstractions" standpoint, curious what do you think about it.

Reply:
If we took a `&'static` reference it would mean any errors returned would need to be defined as `const`s or leaked memory. The `+ 'static` bound here just means the contained data must be owned (or contain only `&'static` references), such that we can hold onto the type for an indeterminate amount of time. https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html

A Box is the best way to take ownership of the arbitrary error type and "erase" it.